### DISCUSSION TITLE
Very hard to build with cmake

### DISCUSSION THREAD

**Discussion by Wim4rk** [musescore/MuseScore#12386] [Q&A] [answered]
Hi,
Trying to build from source. Lots of problems have been solved. The latest one reads like this:
My command: $ cmake -P build.cmake -DCMAKE_BUILD_TYPE=Release

```
/musescore/src/importexport/musicxml/internal/musicxml/musicxmlsupport.h:27:10: fatal error: QAbstractMessageHandler: No such file or directory
   27 | #include <QAbstractMessageHandler>
```
I'm on Ubuntu 20, I use Qt5.15.2.

Is there a Github somewhere? Would somebody hold my hand for the build process?

**Comment by Wim4rk** [comment, marked as answer]
So, missing a library. In case someone has the same problem mentioned above. Install this - not very obvious - package.

`sudo apt-get install libqt5xmlpatterns5-dev`

**Reply by Lukmin1999** [reply]
Is this something that needs to be done for any device or just Ubuntu?
